Source/Core/Common/MathUtil.h

```diff
@@ -30,6 +31,32 @@ constexpr auto Lerp(const T& x, const T& y, const F& a) -> decltype(x + (y - x)
   return x + (y - x) * a;
 }
 
+template <typename Dest, typename T>
+constexpr Dest SaturatingCast(T value)
+{
+  static_assert(std::is_integral<Dest>());
+
+  constexpr Dest lo = std::numeric_limits<Dest>::lowest();
+  constexpr Dest hi = std::numeric_limits<Dest>::max();
+
+  // T being an unsigned integer and Dest unsigned is a problematic case because the value will
```

Comment:
-> T being a signed integer

Reply:
Whoops, good catch. The `value > hi` comparison is also still incorrect, I'll fix this when I can...